### PR TITLE
test: add unit tests for msbuild fix

### PR DESCRIPTION
**Pull request by aniruddhadave**
## Related issue

Closes: #ISSUE_NUMBER_HERE
Add unit tests for MSBuild fix

## Description
Unit Tests Added
vs versions on filesystem -> msbuild path found
1. vs 2022 + vs 2019 -> 2022
2. vs 2019 + 2019 build tools -> vs 2019
3. vs 2017 + 2019 build tools -> vs 2019
4. msbuild 14 -> 14
5. vs 2022 without target + 2019 -> 2019
6. vs 2019 without target and nothing else -> empty string

## Supplemental testing
Describe any testing done in addition to existing unit tests

## Additional context
Please provide any additional information related to this PR

---
*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

File: src/Analysis/Codelyzer.Analysis.Build/WorkspaceBuilderHelper.cs [hunk 1]
```diff
@@ -484,6 +484,7 @@ private EnvironmentOptions GetEnvironmentOptions(bool requiresNetFramework)
             {
                 var msbuildExe = GetFrameworkMsBuildExePath();
                 if (!String.IsNullOrEmpty(msbuildExe)) options.EnvironmentVariables.Add(EnvironmentVariables.MSBUILD_EXE_PATH, msbuildExe);
+                else { throw new Exception(); }
```
Comment (marknfawaz) [inline src/Analysis/Codelyzer.Analysis.Build/WorkspaceBuilderHelper.cs:487]: Can you add some text/description to this exception?

File: tst/Codelyzer.Analysis.Tests/Constants.cs [hunk 2]
```diff
@@ -5,5 +5,13 @@ internal class Constants
         // Do not change these values without updating the corresponding line in .gitignore:  **/Projects/Temp
         // This is to prevent test projects from being picked up in git after failed unit tests.
         internal static readonly string[] TempProjectDirectories = { "Projects", "Temp" };
+
+        internal static string programFiles = "Program Files";
+        internal static string programFilesx86 = "Program Files (x86)";
+        internal static string vs2022MSBuildPath = @"Program Files\Microsoft Visual Studio\2022\Professional\MSBuild\Current\Bin\MSBuild.exe";
+        internal static string vs2019MSBuildPath = @"Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\Current\Bin\MSBuild.exe";
+        internal static string vs2017MSBuildPath = @"Program Files (x86)\Microsoft Visual Studio\2017\Professional\MSBuild\15.0\Bin\MSBuild.exe";
+        internal static string vs2019BuildToolsMSBuildPath = @"Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\MSBuild\Current\Bin\MSBuild.exe";
```
Comment (marknfawaz) [inline tst/Codelyzer.Analysis.Tests/Constants.cs:14]: Why do we need to add these? Aren't we looking in the folder structure anyway?

File: tst/Codelyzer.Analysis.Tests/WorkspaceBuilderHelperTests.cs [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+            Directory.Delete(testMissingTargetsPath, true);
+        }
+
+        [Test]
+        public void TestVS2022()
+        {
+            string actualMsBuildPath = Path.Combine(testvs2022Path, Constants.vs2022MSBuildPath);
+            string msbuildpath = WorkspaceBuilderHelper.GetFrameworkMsBuildExePath(programFilesPath: Path.Combine(testvs2022Path, Constants.programFiles), programFilesX86Path: Path.Combine(testvs2022Path, Constants.programFilesx86));
+            Console.WriteLine(actualMsBuildPath);
+            Console.WriteLine(msbuildpath);
```
Comment (marknfawaz) [inline tst/Codelyzer.Analysis.Tests/WorkspaceBuilderHelperTests.cs:117]: Do we need the console.writeline?

**Review by marknfawaz — CHANGES_REQUESTED** [review]

File: tst/Codelyzer.Analysis.Tests/Constants.cs (hunk 2 above)
Comment (aniruddhadave) [reply tst/Codelyzer.Analysis.Tests/Constants.cs:14]: These are the expected values for test cases. I put them under constants. What we look and get are the actual values.

**Review by marknfawaz — APPROVED** [review]

**Review by saikiranakula-amzn — APPROVED** [review]

**Pull request merged**